swesmith/profiles/rust.py

```
@@ -35,7 +35,6 @@ def dockerfile(self):
 
 RUN git clone https://github.com/{self.mirror_name} /testbed
 WORKDIR /testbed
-RUN {self.test_cmd}
```

If the profiles ran the test command when building the image it might remove the need for a separate `test_build_image.sh` script? Maybe it's annoying when you just want to build the image? An alternative might be to add a flag to create_images to control whether the test command is run?

---

I re-introduced this command and appended a ` || true` to the end.

@richardzhuang0412 correct me if I'm wrong - my guess is that the reason you removed this is because sometimes, there are repositories where not *all*, but a lot of the tests pass. However, because 1+ test fails, the `docker build` comes back with a "failure" because it thinks the image wasn't built successfully?

If this is the case, I think `|| true` could be useful (this is how we're doing it in Golang right now). The command should be considered as run successfully no matter the test outcome.

The problem with this is of course, if the test suite doesn't _actually_ run, and then `|| true` ends up masking that failure condition.
* But to address ^, we can just manually check that `logs/build_images/env/<repo>/build_image.log` reflects the test suite was actually run.

---

An additional point: for some repos, running the test cmd during the initial build is good because that initial run might set up testing infrastructure / download assets, such that the 2nd run and onwards will run much faster.

So if we run testing during the build, when we actually spawn a container and run evaluation, the testing will run much faster (versus if we didn't, every evaluation run would be running tests from scratch, which would take more time).